services/QuillLMS/app/models/activity_topic.rb

```diff
@@ -23,4 +23,10 @@ class ActivityTopic < ApplicationRecord
   belongs_to :topic
 
   validates :topic_id, presence: true
+
```

Comment:
Rails has a [validates_with](https://guides.rubyonrails.org/active_record_validations.html#validates-with) helper that would be more conventional here.  This also establishes a pattern of adding validation error messages that can be useful later.

```
class ActivityTopicLevelOneValidator < ActiveModel::Validator
  def validate(record)
    record.errors.add :base, "Some descriptive error message here" unless record.topic.level_one?
  end
end

class ActivityTopic < ApplicationRecord
  validates_with ActivityTopicLevelOneValidator
end
```